### PR TITLE
[babel-relay-plugin] Move Node ID fieldname definition into forkable file.

### PR DESCRIPTION
**Pull request by alloy**
As discussed in https://github.com/facebook/relay/pull/1232, this should make it easier to maintain a forked version of Relay and change the name of the field to use as Node ID.

@josephsavona Did I understand everything correct?

### CODE REVIEW

File: scripts/babel-relay-plugin/src/RelayQLAST.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ const find = require('./find');
 const invariant = require('./invariant');
 const util = require('util');
 const RelayTransformError = require('./RelayTransformError');
+const { ID } = require('./RelayQLNodeInterface');
```
Comment (josephsavona) [inline scripts/babel-relay-plugin/src/RelayQLAST.js]: nit: no spaces: `const {ID} = ...`

File: scripts/babel-relay-plugin/src/RelayQLPrinter.js [hunk 2]
```diff
@@ -31,6 +31,7 @@ const find = require('./find');
 const invariant = require('./invariant');
 const util = require('util');
 const RelayTransformError = require('./RelayTransformError');
+const { ID } = require('./RelayQLNodeInterface');
```
Comment (josephsavona) [inline scripts/babel-relay-plugin/src/RelayQLPrinter.js]: same here

**Comment by josephsavona** [comment]
@alloy Awesome, thanks! This is exactly what I was thinking.

**Comment by alloy** [comment]
Related but off-topic, why are the compiled plugin sources checked-in but not the runtime ones? I’m wondering if I should do this in my fork, because from my limited experience with NPM it appears that if any `prepublish` work needs to be done you _have_ to push a package to the registry, you can’t use such a package directly from a git repo.

**Comment by josephsavona** [comment]
The plugin sources are checked in compiled because they're used *within* our babel compilation toolchain. It's certainly possible for these to be compiled dynamically anyway (two instances of babel, one compiling plugins for the other), but I *suspect* we avoided that in the RN packager for perf reasons.

**Comment by steveluscher** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-3** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/267584053605947/int_phab) to review.

**Comment by alloy** [comment]
@josephsavona Gotcha. Thanks for all the help 👍 👏

**Comment by josephsavona** [comment]
@alloy thanks for this!

**Comment by alloy** [comment]
Seems like this doesn’t actually yet makes it fully work at runtime. In the case of the root node in a query it works as expected, but otherwise the dataIDs are of the form `client:integer`. Looking into it further, if you have a hunch of where I should look, that would be great.

**Comment by alloy** [comment]
Awww, that was silly. I missed the two places I had actually changed in my previous PR. Will send a PR for that.

**Pull request closed without merging**